Nice! I had no idea about `handleEvent` 👍 I'm down to merge. Looks like there's one lint issue to fix that I spot in the AppVeyor build output:

```
  34:11  error  unnecessary '.call()'  no-useless-call
```

`npm run lint` should show this.

Edit: to fix the lint error, can the `call` be removed? Alternatively, `// eslint-disable-next-line` should do it.

--- a/lib/eventtarget.js
+++ b/lib/eventtarget.js
@@ -28,8 +28,14 @@ function EventTarget() {
         listenerArray = listenerArray.concat(dummyListener);
       }
 
-      for (var i = 0, l = listenerArray.length; i < l; i++) {
-        listenerArray[i].call(self, event);
+      for (var i = 0, l = listenerArray.length, listener; i < l; i++) {
+        listener = listenerArray[i];
+        if (typeof listener === 'object' && typeof listener.handleEvent === 'function') {
+          listener.handleEvent(event);
+        }
+        else {
+          listener.call(self, event);
+        }
       }
     });
   };

